estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -510,55 +525,140 @@ def open_binding(
     binding_index: int,
     state: _ResourceState,
     task: Task,
-    fetch_changes: FetchChangesFn[_BaseDocument] | None = None,
-    fetch_page: FetchPageFn[_BaseDocument] | None = None,
-    fetch_snapshot: FetchSnapshotFn[_BaseDocument] | None = None,
+    fetch_changes: FetchChangesFn[_BaseDocument]
+    | dict[str, FetchChangesFn[_BaseDocument]]
+    | None = None,
+    fetch_page: FetchPageFn[_BaseDocument]
+    | dict[str, FetchPageFn[_BaseDocument]]
+    | None = None,
+    fetch_snapshot: FetchSnapshotFn[_BaseDocument]
+    | dict[str, FetchSnapshotFn[_BaseDocument]]
+    | None = None,
     tombstone: _BaseDocument | None = None,
 ):
     """
     open_binding() is intended to be called by closures set as Resource.open Callables.
 
     It does 'heavy lifting' to actually capture a binding.
 
-    TODO(johnny): Separate into snapshot vs incremental tasks?
+    When fetch_changes, fetch_page, or fetch_snapshot are provided as dictionaries,
+    each function will be run as a separate subtask with its own independent state.
+    The dictionary keys are used as subtask IDs and are used to store and retrieve
+    the state for each subtask in state.inc, state.backfill, or state.snapshot.
     """
 
     prefix = ".".join(binding.resourceConfig.path())
 
     if fetch_changes:
+        if isinstance(fetch_changes, dict):
+            for subtask_id, subtask_fetch_changes in fetch_changes.items():
+
+                async def incremental_subtask(
```

Comment:
Does this work if the closure here just only takes `subtask_task` as an input and doesn't set the `fetch_fn` and `subtask_id` as defaults in the function signature? It took me a while to figure out that it was basically doing the same thing as `closure` in the non-dictionary case, so it might be nice to make the two more symmetrical, unless there's some reason the values need to be initialized like that.

Reply:
Looks good!